lib/hexpm_web/controllers/controller_helpers.ex

```diff
@@ -73,10 +74,16 @@ defmodule HexpmWeb.ControllerHelpers do
 
   defp interpolate_errors(message, opts) do
     Enum.reduce(opts, message, fn {key, value}, message ->
-      if String.Chars.impl_for(key) && String.Chars.impl_for(value) do
-        String.replace(message, "%{#{key}}", to_string(value))
+      pattern = "%{#{key}}"
+
+      if String.contains?(message, pattern) do
+        if String.Chars.impl_for(key) && String.Chars.impl_for(value) do
```

Comment:
This check for key is redundant since it happens after the key is already stringified 2 lines above.

Reply:
Thanks, will fix that in a separate commit!